File: BackendDotNet/Controllers/AuthController.cs
using BackendDotNet.DTOs.Request;
using BackendDotNet.DTOs.Response;
using BackendDotNet.Services;
using Microsoft.AspNetCore.Mvc;

namespace BackendDotNet.Controllers
{
    [ApiController]
    [Route("api/auth")]
    public class AuthController : ControllerBase
    {
        private readonly IAuthService _authService;

        public AuthController(IAuthService authService)
        {
            _authService = authService;
        }

        [HttpPost("register/donor")]
        public async Task<ActionResult<string>> RegisterDonor([FromBody] DonorRegisterRequest registerRequest)
        {
            try
            {
                var response = await _authService.RegisterDonorAsync(registerRequest);
                return StatusCode(201, response);
            }
            catch (InvalidOperationException ex)
            {
                return BadRequest(ex.Message);
            }
            catch (Exception ex)
            {
                return StatusCode(500, $"Internal server error: {ex.Message}");
            }
        }

        [HttpPost("register/hospital")]
        public async Task<ActionResult<string>> RegisterHospital([FromBody] HospitalRegisterRequest registerRequest)
        {
            try
            {
                var response = await _authService.RegisterHospitalAsync(registerRequest);
                return StatusCode(201, response);
            }
            catch (InvalidOperationException ex)
            {
                return BadRequest(ex.Message);
            }
            catch (Exception ex)
            {
                return StatusCode(500, $"Internal server error: {ex.Message}");
            }
        }

        [HttpPost("login")]
        public async Task<ActionResult<JwtAuthResponse>> Login([FromBody] LoginRequest loginRequest)
        {
            try
            {
                var response = await _authService.LoginAsync(loginRequest);
                return Ok(response);
            }
            catch (UnauthorizedAccessException ex)
            {
                return Unauthorized(ex.Message);
            }
            catch (Exception ex)
            {
                return StatusCode(500, $"Internal server error: {ex.Message}");
            }
        }
    }
}

File: backend spring/src/main/java/com/bloodbank/controller/AuthController.java
package com.bloodbank.controller;

import com.bloodbank.dto.request.DonorRegisterRequest;
import com.bloodbank.dto.request.HospitalRegisterRequest;
import com.bloodbank.dto.request.LoginRequest;
import com.bloodbank.dto.response.JwtAuthResponse;
import com.bloodbank.service.AuthService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/auth")
public class AuthController {

    @Autowired
    private AuthService authService;

    @PostMapping("/register/donor")
    public ResponseEntity<String> registerDonor(@RequestBody DonorRegisterRequest registerRequest) {
        String response = authService.registerDonor(registerRequest);
        return new ResponseEntity<>(response, HttpStatus.CREATED);
    }

    @PostMapping("/register/hospital")
    public ResponseEntity<String> registerHospital(@RequestBody HospitalRegisterRequest registerRequest) {
        String response = authService.registerHospital(registerRequest);
        return new ResponseEntity<>(response, HttpStatus.CREATED);
    }

    @PostMapping("/login")
    public ResponseEntity<JwtAuthResponse> login(@RequestBody LoginRequest loginRequest) {
        JwtAuthResponse response = authService.login(loginRequest);
        return ResponseEntity.ok(response);
    }
}